dandiapi/api/models/version.py

```diff
@@ -90,18 +90,25 @@ def asset_validation_errors(self) -> list[str]:
             status=Asset.Status.VALID
         ).values('status', 'path', 'validation_errors')
 
-        # Aggregate errors
+        asset_validating_error = {
+            'field': '',
+            'message': 'asset is currently being validated, please wait.',
+        }
+
+        def inject_path(asset: dict, err: dict):
+            return {**err, 'path': asset['path']}
+
+        # Aggregate errors, ensuring the path of the asset is included
         errors = []
         for asset in invalid_assets:
```

Comment:
I think this will be really slow and/or memory intensive if a dandiset has hundreds of thousands of invalid assets. In general, I don't like how we go about storing asset validation errors in the database, but this PR is a strict improvement over the current situation imo. Perhaps related to #1214.